### ISSUE TITLE
pretext + kmeans

### ISSUE THREAD

**Issue by mynameischaos** [wvangansbeke/Unsupervised-Classification#38]
Hi~ I am following your beautiful work, but I met some questions and hope you can help me. 

I am using your pretext model of cifar10(https://drive.google.com/file/d/1Cl5oAcJKoNE5FSTZsBSAKLcyA5jXGgTT/view) to require the 128-dimension features and run kmeans algorithm, I got very low performance than your paper.  (acc: 20% vs 60%). But I first run TSNE on 128-dimension features and got 2-dimension new features and used kmeans algorithm, I got the similar accuracy with your paper.  Do you meet the same question?

**Comment by wvangansbeke** [comment]
Hi @mynameischaos,

You don't have to perform a TSNE. I simply did Kmeans on the output features for cifar10, since it performed slightly better than on the normalized features before the projection head (i.e. 512 dimensional features). For ImagNet, using the features before the projection head is slightly better though. Normally you can simply use the Kmeans implementation from FAISS (on the GPU) or the (minibatch) Kmeans implementation from sklearn (on the CPU). You should get around 65% for CIFAR10. 

Hope this helps.

**Comment by wvangansbeke** [comment]
If you have more questions please let me know. I'm closing this for now.

**Comment by mynameischaos** [comment]
Thank you for your reply. But I follow your suggestion. I downloaded your provided pre-trained model. (https://drive.google.com/file/d/1Cl5oAcJKoNE5FSTZsBSAKLcyA5jXGgTT/view) and performed kmeans on features(10000 * 128) by sklearn or faiss. The results are much lower than 65%. Can you provide your code for this experiment? Thank you very much.

**Comment by mynameischaos** [comment]
@wvangansbeke

**Comment by linqinghong** [comment]
@mynameischaos Hello, have you find the correct way?

**Comment by TsungWeiTsai** [comment]
I guess you can try to cluster the training + testing set together and evaluate the clustering performance on the testing set to see if it matches the performance reported in their paper. Also, you can try spherical k-means which will provide better performance on the l2-normalized data, as shown in [ICLR2021 MiCE](https://openreview.net/pdf?id=gV3wdEOGy_V).

**Comment by linqinghong** [comment]
many thank, I will try!

----------

该邮件从移动设备发送


--------------原始邮件--------------
发件人："Tsai Tsung Wei "<notifications@github.com&gt;;
发送时间：2021年2月28日(星期天) 晚上9:30
收件人："wvangansbeke/Unsupervised-Classification" <Unsupervised-Classification@noreply.github.com&gt;;
抄送："Vincent "<linqinghong@email.szu.edu.cn&gt;;"Comment "<comment@noreply.github.com&gt;;
主题：Re: [wvangansbeke/Unsupervised-Classification] pretext + kmeans (#38)
-----------------------------------

 

 
I guess you can try to cluster the training + testing set together and evaluate the clustering performance on the testing set to see if it matches the performance reported in their paper. Also, you can try spherical k-means which will provide better performance on the l2-normalized data, as shown in ICLR2021 MiCE.
